newIDE/app/src/EventsFunctionsExtensionEditor/OptionsEditorDialog/UsersAutocomplete.js

```diff
@@ -0,0 +1,153 @@
+//@flow
+import * as React from 'react';
+import { useState } from 'react';
+import { t } from '@lingui/macro';
+import SemiControlledMultiAutoComplete from '../../UI/SemiControlledMultiAutoComplete';
+import {
+  searchUserPublicProfilesByUsername,
+  type UserPublicProfile,
+  getUserPublicProfilesByIds,
+} from '../../Utils/GDevelopServices/User';
+
+import useForceUpdate from '../../Utils/UseForceUpdate';
+
+type Props = {|
+  userIds: gdVectorString,
+  floatingLabelText?: React.Node,
+  helperText: React.Node,
+|};
+
+type Option = {|
+  text: string,
+  value: string,
+|};
+
+const getErrorMessage = (error: ?Error) => {
+  if (error) return 'Error while loading users';
+};
+
+export const UsersAutocomplete = (props: Props) => {
+  const forceUpdate = useForceUpdate();
+  const [users, setUsers] = React.useState<Array<Option>>([]);
+  const [userInput, setUserInput] = useState('');
+  const [loading, setLoading] = useState(false);
+  const [
+    completionUserPublicProfiles,
+    setCompletionUserPublicProfiles,
+  ] = React.useState<Array<UserPublicProfile>>([]);
+  const [error, setError] = React.useState(null);
+
+  // Recalculate if the userInput has changed.
+  const searchUserPublicProfiles = React.useCallback(
+    async () => {
+      setError(null);
+      if (!userInput) {
+        setCompletionUserPublicProfiles([]);
+        return;
+      }
+      try {
+        setLoading(true);
+        const userPublicProfiles = await searchUserPublicProfilesByUsername(
```

Comment:
We seem to have the lodash.debounce dependency, you could use it to reduce the api calls.
https://stackoverflow.com/a/62017005/9517183

Reply:
good catch, I meant to use it and forgot!